### ISSUE TITLE
serialize concurrent supervisord commands that change state (start, stop, restart)

### ISSUE THREAD

**Issue by sanikolov** [Supervisor/supervisor#1232]
Suppose that several restart commands are issued in parallel via supervisorctl as illustrated below
`# supervisorctl -c /path/to/supervisord.conf restart all & supervisorctl -c /path/to/supervisord.conf restart all & supervisorctl -c /path/to/supervisord.conf restart all &`

Currently (latest versions) these commands may result in a mess, some old process hierarchies mixed with new ones, some processes will be down, other random ones will be up, no consistency, dependencies may not be satisfied.

Proposed behaviour - supervisord receives change requests such as stop, start, restart and executes them serially, only after the previous command has fully completed, whereas read only commands like status may be satisfied immediately.

The scenario described here (concurrent change requests) is not really contrived. For instance, you may have daemon1 and daemon2 which may issue restarts without coordination with one another.

**Comment by mnaberez** [comment]
I am closing this issue to merge it into #122.  The [original comment](https://github.com/Supervisor/supervisor/issues/122#issue-4827059) in #122 is requesting that actions be serialized, as is requested here.

**Comment by collinbachi** [comment]
But #122 has been open for 8 years with no resolution, so, this is basically closed forever?

**Comment by mnaberez** [comment]
> so, this is basically closed forever?

This issue was merged into #122 because they are basically the same issue as noted above.  We generally do not leave duplicates open in this repository so that discussions can be focused on one ticket instead of spread across the duplicates.
